frontend/src/components/CourseCard.js

```diff
@@ -1,39 +1,48 @@
 import React, { Component } from 'react';
 import $ from 'jquery';
+import { Draggable } from 'react-beautiful-dnd';
+import { EXTERNAL_CREDITS_SEMESTER_INDEX } from 'utils/SemesterUtils';
 
 export default class CourseCard extends Component {
   removeCourse = () => {
     this.props.onCourseRemove(this.props.semIndex, this.props.courseIndex);
     $(`#${this.props.courseKey}`).popover('hide');
   };
 
-  getClassNames = () => {
+  getClassNames = (isDragging) => {
     let classNames = [];
     classNames.push('course-card');
     classNames.push(this.props.course['semester']);
-    if (this.props.isDragging) classNames.push('course-card-shadow');
-    if (this.props.isPlaceholder) classNames.push('placeholder-course');
+    if (isDragging) classNames.push('course-card-shadow');
     return classNames.join(' ');
   };
 
   render() {
-    let course = this.props.course;
+    let { course, courseKey, courseIndex, semIndex } = this.props;
     return (
       <div className="unbreakable">
-        <div
-          id={this.props.courseKey}
-          title={course['name']}
-          className={this.getClassNames()}
-          ref={this.props.innerRef}
-          {...this.props.draggable}
-          {...this.props.dragHandle}
+        <Draggable
```

Comment:
This Draggable component is used by react-beautiful-dnd, and needs to wrap components that are "draggable". I moved it into this CourseCard component because all CourseCards are basically draggable (except the external credits course cards), so this reduces repetition. Notice it's now gone from Semester.js where it was being repeated for each CourseCard.